### PR TITLE
Replace BSAggregate

### PR DESCRIPTION
**Pull request by j-lawson**

### CODE REVIEW

File: R/aggregate.R [hunk 1]
```diff
@@ -1,155 +1,96 @@
...
+# "regionGroupID" refers to which GRanges from regionsGRL the given
+# region was a member of. "withinGroupID" refers to the index of the given
+# region within its GRanges object.
+# "regionID" has the index for the given region as if all the GRanges from
+# regionsGRL were combined into a single GRanges object
+# The "N" column has the counts for number of query regions 
+# overlapping with that regionsGRL region
+calcOLCount = function(queryRegionDT, regionsGRL) {
+    jExpr = ".N"
+    BSDT = queryRegionDT
```
Comment (nsheff) [inline R/aggregate.R:26]: probably best to rename this variable

**Comment by j-lawson** [comment]
It's much less confusing because it has 2 parameters instead of 10, 8 of which were not being used. Better for package maintenance.

**Comment by nsheff** [comment]
but... it duplicates 80% of the functionality of the other function. Why not just write a partial function with the signature you want (the same signature as your new function), but have it call the BSAggregate function with fixed arguments ?

This way, you only have to maintain this code in one place. Ideally, the BSAggregate function is in some other package and we just call it from there, since its utility spans this package.

**Comment by j-lawson** [comment]
I don't think that version of the BSAggregate function is in any other public package

**Comment by nsheff** [comment]
Right, that makes sense. Well, I think this is fine then. I think eventually, if we ever have this aggregate function in generic form somewhere in some package, then it would make sense to switch over to using just the parts of that. Or, maybe at some point it will make sense for the full function to be part of this package, and the other packages can then just depend on this one. What do you think?

**Comment by j-lawson** [comment]
Yeah that is understandable. Maintaining that function has required a lot of time so I'm not sure that it is worth it now and may not be in the future but it just depends on the situation I guess

File: R/aggregate.R (hunk 1 above)
Comment (j-lawson) [reply R/aggregate.R]: changed

**Comment by j-lawson** [comment]
I'm getting a warning when using devtools::document() that 

"In setup_ns_exports(path, export_all, export_imports) :
  Objects listed as exports, but not present in namespace: BSAggregate"

I couldn't find a setup_ns_exports function though. Any idea why this is happening? The BSAggregate Rd file has been deleted and the NAMESPACE file updated.

**Comment by nsheff** [comment]
document should manage the namespace file for you... you don't need to change it.

**Comment by nsheff** [comment]
did you ever solve this?

**Comment by j-lawson** [comment]
I'm not sure where the warning came from but I think the changes should be fine.

**Pull request merged**